core/src/main/scala/knobs/AWS.scala

```diff
@@ -36,59 +38,67 @@ object aws {
-    Config.parse(s"$section { $field = ${formatter(response)} }").fold(Task.fail, Task.now)
+  private def convert[F[_]](field: String, section: String = "aws", formatter: String => String = defaultFormating)(response: String)(implicit F: MonadError[F, Throwable]): F[Config] =
+    Config.parse(s"$section { $field = ${formatter(response)} }").fold(F.raiseError, F.pure)
 
-  def userdata: Task[Config] = fetch("user-data")
-    .flatMap(response => Config.parse(response)
-    .fold(Task.fail, Task.now)) or Task.now(Config.empty)
+  def userdata[F[_]](implicit F: Sync[F]): F[Config] =
+    or(
+      fetch("user-data").flatMap(response => Config.parse(response).fold(F.raiseError, F.pure)),
+      F.pure(Config.empty)
+    )
 
-  def instance: Task[Config] =
-    fetch("meta-data/instance-id").flatMap(convert("instance-id"))
+  def instance[F[_]: Sync]: F[Config] =
+    fetch("meta-data/instance-id").flatMap(convert[F]("instance-id"))
 
-  def ami: Task[Config] =
-    fetch("meta-data/ami-id").flatMap(convert("ami-id"))
+  def ami[F[_]: Sync]: F[Config] =
+    fetch("meta-data/ami-id").flatMap(convert[F]("ami-id"))
 
-  def securitygroups: Task[Config] = fetch("meta-data/security-groups").flatMap(
-    convert("security-groups", formatter = x => "[" + x.split('\n').map(x => "\""+x+"\"").mkString(", ") + "]"  ))
+  def securitygroups[F[_]: Sync]: F[Config] = fetch("meta-data/security-groups").flatMap(
+    convert[F]("security-groups", formatter = x => "[" + x.split('\n').map(x => "\""+x+"\"").mkString(", ") + "]"  ))
 
-  def zone: Task[Config] =
-    fetch("meta-data/placement/availability-zone").flatMap(convert("availability-zone"))
+  def zone[F[_]: Sync]: F[Config] =
+    fetch("meta-data/placement/availability-zone").flatMap(convert[F]("availability-zone"))
 
-  def region: Task[Config] =
-    fetch("meta-data/placement/availability-zone").map(_.dropRight(1)).flatMap(convert("region"))
+  def region[F[_]: Sync]: F[Config] =
+    fetch("meta-data/placement/availability-zone").map(_.dropRight(1)).flatMap(convert[F]("region"))
 
-  def localIP: Task[Config] =
-    fetch("meta-data/local-ipv4").flatMap(convert("local-ipv4", section = "aws.network"))
+  def localIP[F[_]: Sync]: F[Config] =
+    fetch("meta-data/local-ipv4").flatMap(convert[F]("local-ipv4", section = "aws.network"))
 
-  def publicIP: Task[Config] =
-    fetch("meta-data/public-ipv4").flatMap(convert("public-ipv4", section = "aws.network"))
+  def publicIP[F[_]: Sync]: F[Config] =
+    fetch("meta-data/public-ipv4").flatMap(convert[F]("public-ipv4", section = "aws.network"))
 
   /**
    * Overhead for calling this mofo is high, so its not the kind of
    * thing you want to call anymore than once, but its value is cached internally
    * within the Config object that this is composed with.
    */
-  lazy val config: Task[Config] =
-    (for {
+  def config[F[_]](implicit F: Sync[F]): F[Config] =
```

Comment:
Is the comment above still true because given the change from val to def?

Reply:
It should be the running of the effect, not the construction of the effect, that's expensive.  So I think the comment is still valid.  You wouldn't want to run this on, say, every request.  At a server startup, sure.

Reply:
yup, that makes sense